hermes-api/src/main/java/pl/allegro/tech/hermes/api/OAuthProviderWithOptionalSocketTimeout.java

```diff
@@ -0,0 +1,43 @@
+package pl.allegro.tech.hermes.api;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Optional;
+
+public class OAuthProviderWithOptionalSocketTimeout {
```

Comment:
do we really need this class (a class with such explicit name)? :) I mean, can't we just add the optional socket timeout to the existing one?

Reply:
This class is only needed to be able to fetch existing zookeeper saved oauth providers. Its implementation detail. BTW this class should be placed in infrastructure package.

Reply:
done